### DISCUSSION TITLE
Reliable way to convert evaluated expression to TeX

### DISCUSSION THREAD

**Discussion by ovk** [josdejong/mathjs#2727] [Ideas]
While MathJS Node can be easily converted to TeX by simply calling `toTex` method, it looks like there's no reliable way to achieve the same result for the calculated value. While some values may have `toTex` method, the others don't.

Please consider the following examples:

- If the result of evaluation is number, string, boolean, fraction, complex number, matrix - the resulting object won't have `toTex` method.
- If the result is a custom type, it may have `toTex` method if it was originally registered for this custom type.
- If the result is expression itself (e.g. value of `simplify("pi + 2pi")`), the resulting object will have `toTex` method.

I think it would be nice to be able to convert the result of evaluation to TeX as well. For example, if you are multiplying two matrices and converting this expression to TeX, you are very likely want to convert the resulting matrix to TeX as well.

So far my only idea of how to achieve this, is to feed the result back to `parse()` method and then convert this Node to TeX. But there's should be a better way.

**Comment by josdejong** [comment]
Currently, the `.toTex()` methods are only available in the context of parsed expressions, these methods are indeed not available on any of the data types (matrix, complex number, etc.). 

It could be interesting to implement `toTex()` methods for all data types too, so yo could do [pretty printing](http://mathjs.org/examples/browser/pretty_printing_with_mathjax.html) on evaluated results too.  I think this should be implemented in the form of a function `math.toTex(x)`, since we cannot add `.toTex()` methods to basic types like number and boolean (well, you can in JS but it's no good practice). For the data classes that math.js controls (like Matrix) it could be implemented as a `.toTex()` method on the class. 

Anyone interested in implementing toTex for all data types?
